homework-submission/src/App.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import "./App.css";
+import todosJSON from "./Sources/todosJSON.json";
+import List from "./Components/List";
+import NewTodoForm from "./Components/NewTodoForm";
+import uuid from "uuid/v4";
+
+const TODO_TEMPLATE = {
+  description: "",
+  deadline: null,
+  id: null,
+  done: false
+};
+
+export default class App extends Component {
+  state = {
+    todos: [...todosJSON],
+    updatedTodo: { ...TODO_TEMPLATE },
+    newTodo: {
+      ...TODO_TEMPLATE
+    }
+  };
+
+  handleToggleCheck = item => {
+    const todos = [...this.state.todos];
+    const todosCopy = todos.map(todo => {
+      if (todo.id !== item.id) return todo;
+      return { ...todo, done: !todo.done };
+    });
+
+    this.setState({ todos: todosCopy });
+  };
+
+  handleNewDescription = e => {
+    const newTodo = { ...this.state.newTodo };
+    newTodo.description = e.target.value;
+
+    this.setState({ newTodo });
+  };
+
+  handleNewDeadline = e => {
+    // is using a ref better here ? then combining this and handeleNewDescription in one go
+    let newTodo = { ...this.state.newTodo };
+    newTodo.deadline = e.target.value;
+
+    this.setState({ newTodo });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    let newTodo = { ...this.state.newTodo };
+    if (!newTodo.description || !newTodo.deadline) return;
+
+    let { todos } = this.state;
+    newTodo = { ...newTodo, id: newTodo.id || uuid() };
+    todos = [...todos, newTodo];
+
+    newTodo = TODO_TEMPLATE;
+
+    this.setState({ todos, newTodo });
+  };
+
+  handleRemove = itemID => {
+    const { todos } = this.state;
+    const newTodos = todos.filter(todo => todo.id !== itemID);
+
+    this.setState({ todos: newTodos });
+  };
+
+  handleEdit = itemID => {
+    let updatedTodo = { ...this.state.updatedTodo };
```

Comment:
you are copying `updateTodo` from state - good! - but then you are overwriting it 3 lines below with `updateTodo = entity`. Is there a reason I don't see?
Instead of calling it `entity` you can call it directly `updateTodo` there, `find` returns a copy of the object, it's immutable by default 🎉 

It would become:
```js
const todos = [...this.state.todos];
const updatedTodo = todos.find(todo => todo.id === itemID);
this.setState({ updatedTodo });
```

Reply:
i don't know why i over complicate things, must do a zen meditation before starting to code 😄